### DISCUSSION TITLE
Audio track meta info gets lost via RTSP

### DISCUSSION THREAD

**Discussion by p1ngb4ck** [bluenviron/mediamtx#508] [Questions]
Hey =)

Thanks so much for your awesome work ! Appreciate it a lot and works great so far - and I am working on a big setup with around 50 Full-HD Cams attached to a dvb-modulator => tvheadend => rtsp-simple-server proxy > 200 Viewers.
Currently I have a tiny problem though, maybe sb can shed some light into this for me :) :

I run OnDemand sources using ffmpeg, which grabs an http-stream from tvheadend and publishes this one via rtsp to rtsp-simple-server. Unfortunately an RTSP-client does not get any "meta" info about the audio-stream and can therefore not automatically select the correct one based on his own defaults (obey the client`s preffered audio language settings).
Am I doing sth wrong, or is this a required enhancement for the future?

Thanks a bunch & best regards,
Oliver

**Comment by aler9** [comment]
Hello, my guess is that meta is probably present in the SDP (description) of the stream, but at the moment rtsp-simple-server filters out most of the SDP in order to homogenize outgoing streams, including language information.

If you provide the SDP of the original stream, we can patch rtsp-simple-server not to filter out the language information. You can obtain the SDP by setting `logLevel: debug` inside the configuration file and check for a `DESCRIBE` response, like

```
CSeq: 3
Content-Base: rtsp://xx/Streaming/Channels/103/
Content-Length: 607
Content-Type: application/sdp

v=0
o=- 1629297897201241 1629297897201241 IN IP4 192.168.1.24
s=Media Presentation
e=NONE
b=AS:5050
t=0 0
a=control:rtsp://xx:554/Streaming/Channels/103/
m=video 0 RTP/AVP 96
c=IN IP4 0.0.0.0
b=AS:5000
a=recvonly
a=x-dimensions:1920,1080
a=control:rtsp://xx:554/Streaming/Channels/103/trackID=1
a=rtpmap:96 H264/90000
a=fmtp:96 profile-level-id=420029; packetization-mode=1; sprop-parameter-sets=Z2QAKqwsaoHgCJ+WbgICAoAAA4QAAK/IAg==,aO48sA==
a=Media_header:MEDIAINFO=494D4B48010300000400000100000000000000000000000000000000000000000000000000000000;
a=appversion:1.0
```

**Reply by p1ngb4ck** [reply]
Hi, thanks for your answer !

My log gives these info :
2021/08/19 12:37:18 D [1/0] [path proxytest] [rtsp source] s->c RTSP/1.0 200 OK
CSeq: 2
Content-Base: rtsp://10.191.0.1:8554/teststream/
Content-Length: 640
Content-Type: application/sdp
Server: gortsplib

v=0
o=- 0 0 IN IP4 127.0.0.1
s=Stream
c=IN IP4 0.0.0.0
t=0 0
m=video 0 RTP/AVP 96
a=rtpmap:96 H264/90000
a=fmtp:96 packetization-mode=1; sprop-parameter-sets=Z01AKJZSgPAIn7wGhAgICQAAAwABAAADADKU,aO88gA==; profile-level-id=4D4028
a=control:trackID=0
m=audio 0 RTP/AVP 97
b=AS:192
a=rtpmap:97 MPEG4-GENERIC/48000/6
a=fmtp:97 profile-level-id=1;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3; config=11B0
a=control:trackID=1
m=audio 0 RTP/AVP 98
b=AS:192
a=rtpmap:98 MPEG4-GENERIC/48000/6
a=fmtp:98 profile-level-id=1;mode=AAC-hbr;sizelength=13;indexlength=3;indexdeltalength=3; config=11B0
a=control:trackID=2
</CODE>

**Reply by aler9** [reply]
The one you posted is the SDP sent from the server to the client, i.e:

tvheadend -----> rtsp-simple-server ----HERE----> client

i need the one before it, this:

tvheadend ---HERE--> rtsp-simple-server --------> client

**Reply by p1ngb4ck** [reply]
Hmm actually I got that before, that you need the info from the publishing-stream and tried to provide that one.
This is my current setup:
Server1 (TVHeadend) <= FFMPEG on Server 2 => RTSP => RTSP-Simple-Server on Server2
FFMPeg on server2 receives a http input stream and pipes it to rtsp-simple-server via RTSP. My log snippet was from the publishing stream created by FFMPeg (at least I think so). For testing purposes, I disabled anything except one publisher path and one proxy for it - and re-checked the logs for the DESCRIBE section - looks exactly similar

Maybe FFMPeg doesnt pass the correct SDP info already?

**Reply by aler9** [reply]
> Maybe FFMPeg doesnt pass the correct SDP info already?

Yes, unfortunately this is the situation.

**Reply by p1ngb4ck** [reply]
Thanks for your help !

[a comment by an automated account is not shown]
